test/handlers/admin/data-contributions.js

```diff
@@ -81,12 +81,19 @@ describe('admin data contributions handler', () => {
     });
 
     it('updates the DataContribution and redirects to /admin/data-contributions', () => {
-      const payload = {
-        approved: true,
-        curation_comments: 'My comments',
-      };
+      let payload;
       let dataContributionId;
 
+      factory.create('dataCategory')
+        .then((dataCategory) => {
+          payload = {
+            approved: true,
+            trial_id: '11111111-1111-1111-1111-111111111111',
+            data_category_id: dataCategory.id,
+            curation_comments: 'My comments',
+          };
+        });
```

Comment:
There's a race condition here. As these promises aren't part of the promises we return, there's no guarantee that they would be executed in order. If they take too long, the test can even finish before they finish executing. To solve this, you must move it to the returned promises.

You could add it between the `.then((dataContribution) => (dataContributionId = dataContribution.attributes.id))` and the `server.inject` lines, simply creating the new data category and changing the `payload` with its ID.